src/asciidoc/docs/guidelines/frontendUploading.adoc

```diff
@@ -0,0 +1,148 @@
+* Backend communication and the <<withkoji-database-package,@withkoji/database-sdk>> package.
+* Performing HTTP requests (this example uses https://github.com/axios/axios[axios]).
+
+== Configuring the backend
+
+To upload from the frontend, you first need to acquire a presigned URL by using the Koji database SDK.
+
+In a backend service for your Koji project, set up an Express.js server with a POST method to the `/get-upload-url` endpoint.
+From this endpoint, call the following function.
+
+[source, javascript]
+----
+import Database from '@withkoji/database';
+
+const getUploadURL = async (req, res) => {
+  try {
+    const { filename } = req.body;
+
+    const db = new Database({
+      projectId: res.locals.KOJI_PROJECT_ID,
+      projectToken: res.locals.KOJI_PROJECT_TOKEN,
+    });
+
+    const data = await db.generateSignedUploadRequest(filename);
+
+    res.status(200).json({ data });
+  } catch (err) {
+    console.log('err', err);
+    res.status(500).json({ err: err.toString() });
+  }
+};
+
+export default getUploadURL;
+----
+
+== Configuring the frontend
+
+From your frontend service, call the `/get-upload-url` endpoint to retrieve the presigned URL.
+For more information on the response value see <<#_signed_request_response>>.
+
+[source,javascript]
+----
+const { data: { data: { signedRequest, url } } = {} } = await axios({
+  method: 'post',
+  url: `${serviceMap.backend}/get-upload-url`,
+  data: {
+    filename: file.name,
+  },
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+----
+
+Create form data and include the presigned URL data and the file object to be uploaded.
+
+[source,javascript]
+----
+// Build form data to send to s3
+const formData = new FormData();
+
+// Attach each of the request fields to the form
+Object.keys(signedRequest.fields).forEach((key) => {
+  formData.append(key, signedRequest.fields[key]);
+});
+
+// Append file last
+formData.append('file', file);
+----
+
+Post this data to the URL, and then check for a successful status (since there is no response data).
+If the status is `204`, you can now use `signedRequest.url` which will point to your uploaded file.
+
+[source,javascript]
+----
+// Post to s3
+const { status } = await axios({
+  headers: { 'content-type': 'application/x-www-form-urlencoded' },
+  method: 'POST',
+  data: formData,
+  url: signedRequest.url,
+});
+
+if (status === 204) {
+  const image = new window.Image()
+  image.onload = () => {
+    document.body.append(image);
+  }
+  image.src = signedRequest.url;
+}
+----
+
+== Signed request response
+
+As the `generateSignedUploadRequest` method is a wrapper for the https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-authentication-HTTPPOST.html[presigned upload request in Amazon S3] the object returned from the `get-upload-url` endpoint is equivalent, all the data stored within the `signedRequest` property are related to the Amazon Web Services functionaliy.
```

Comment:
```suggestion
As the `generateSignedUploadRequest` method is a wrapper for the https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-authentication-HTTPPOST.html[presigned upload request in Amazon S3], the return objects are equivalent.
The data in the `signedRequest` property relates to the Amazon Web Services functionality.
```

Fixed a typo and tried to make this a little shorter. Might not be the best rewriting so feel free to edit again.